users/views.py

```diff
@@ -1,3 +1,143 @@
+#                 request.user.skill_1 = Skill.objects.get(skill=request.POST.get('skill_1'))
+#             except Skill.DoesNotExist:
+#                 request.user.skill_1 = None
+#             try:
+#                 request.user.skill_2 = Skill.objects.get(skill=request.POST.get('skill_2'))
+#             except Skill.DoesNotExist:
+#                 request.user.skill_2 = None
+#             try:
+#                 request.user.skill_3 = Skill.objects.get(skill=request.POST.get('skill_3'))
+#             except Skill.DoesNotExist:
+#                 request.user.skill_3 = None
+#             request.user.save()
+#             return render(request, 'users/test.html', {})
+#         else:
+#             return render(request, 'users/register_skills.html', {})
+
+
+def view_profile(request, pk):
+    if not request.user.is_authenticated:
+        return render(request, 'users/login.html', {})
+    else:
+        user = get_object_or_404(CustomUser, pk=pk)
+        return render(request, 'users/profile.html', {'user': user})
+
+
+def update_profile(request, pk):
+    if not request.user.is_authenticated:
+        return render(request, 'users/login.html', {})
+    elif request.user.pk != pk:
+        return render(request, 'users/login.html', {})
+    else:
+        if request.method != 'POST':
+            return render(request, 'users/update_profile.html', {})
+        else:
+            request.user.first_name = request.POST.get('first_name')
+            request.user.last_name = request.POST.get('last_name')
+            mobile = request.POST.get('mobile')
+            sap_id = request.POST.get('sap_id')
+            if CustomUser.objects.filter(mobile=mobile)[0].id != pk:
+                error = 'The mobile number is already in use by another account.'
+                return render(request, 'users/update_profile.html', {'error': error})
+            elif CustomUser.objects.filter(sap_id=sap_id)[0].id != pk:
+                error = 'The SAP ID is already in use by another account.'
+                return render(request, 'users/update_profile.html', {'error': error})
+            request.user.photo = request.FILES.get('photo', None)
+            request.user.bio = request.POST.get('bio')
+            request.user.year = request.POST.get('year')
+            try:
+                request.user.skill_1 = Skill.objects.get(skill=request.POST.get('skill_1'))
+            except Skill.DoesNotExist:
+                request.user.skill_1 = None
+            try:
+                request.user.skill_2 = Skill.objects.get(skill=request.POST.get('skill_2'))
+            except Skill.DoesNotExist:
+                request.user.skill_2 = None
+            try:
+                request.user.skill_3 = Skill.objects.get(skill=request.POST.get('skill_3'))
+            except Skill.DoesNotExist:
+                request.user.skill_3 = None
+            request.user.save()
+            return render(request, 'users/profile.html', {'user': request.user})
```

Comment:
End with a new line. 
Also a suggestion, you don't have to explicitly pass pk while referring to different user pages, the user is logged in using a session. So the first check `if request.user.is_authenticated():`  is sufficient. You can access any parameter of the `user` object directly.  This is applicable in the front end as well. For example you'll want the email near the **Visit Profile** button, just use `request.user.email`

Reply:
I already ended with a new line. Also, the reason I am taking the pk as primary key is to check that the user who is supposed to view the page is the one actually doing so. For example, if I am logged in to the website, I should not be able to type "www.skillfinder.com/profile/3/update" and update the profile if I am not the user with id = 3.

Reply:
So don't pass your id in a GET request. You always use a POST Request. Never use such sensitive information in the URL.

Reply:
I am not passing any GET requests. This is just how the URL has been formed. Each user profile has a URL corresponding to the user id. Since we are anyways using email as username, I don't see how this could be an issue.

Reply:
I've no issues with it. @AvaisP just verify once.

Reply:
line 119 does not reload the page and when i reload it again sends a post request. Try using redirect.

Reply:
It does take the user back to the profile page, although the URL does not change, only the profile page gets rendered. I could do it using redirect as well, if you want.

Reply:
This applies for the other renders as well. I don't mind it either way. Whatever you guys prefer.

Reply:
Change the renders to redirect wherever it can be done easily. We can bear the rest for now. And use redirect instead of render wherever you can from now on.

Reply:
I have updated the views as you asked. Now, the code uses render only when necessary.

Reply:
Why didn't you change line 119 to redirect as well. Is it  going to break something?